### DISCUSSION TITLE
Build Stellarium from source with support for ShowMySky

### DISCUSSION THREAD

**Discussion by Atque** [Stellarium/stellarium#3142] [Q&A] [answered]
I can't really wrap my head around how to build Stellarium with support for the ShowMySky atmosphere model. I get no errors while configuring the project in QtCreator, but when running after building, the following message appears:

`08:00:22: Starting C:\Devel-DIVERSE\build-stellarium-master_Qt_6_4_2_MSVC2019_64bit-Release\_deps\showmysky-qt6-build\CalcMySky\calcmysky.exe...`
`08:00:22: The process failed to start. Either the invoked program "C:\Devel-DIVERSE\build-stellarium-master-Desktop_Qt_6_4_2_MSVC2019_64bit-Release\_deps\showmysky-qt6-build\CalcMySky\calcmysky.exe" is missing, or you may have insufficient permissions to invoke the program.`

I can, however, start the `stellarium.exe` from the build directory. However, ShowMySky won't work, as the module is not found.

How do I successfully build Stellarium with ShowMySky support?

**Comment by alex-w** [comment, marked as answer]
You tried run calcmysky after building Stellarium. 

![2023-03-30_13-10-54](https://user-images.githubusercontent.com/88731/228745693-108346d0-b1eb-4b4d-886f-630c21dcd291.png)
Sorry for Russian UI, but you need to change the target in third column

**Reply by Atque** [reply]
Thanks, that works, but it still won't let me use the ShowMySky model. It says a module is missing?
![bild](https://user-images.githubusercontent.com/37028368/228748150-3e323208-ee60-4a12-832d-1469723d2fde.png)

**Reply by 10110111** [reply]
`ShowMySky-Qt6.dll` should be either near `stellarium.exe`, or in a location specified in `PATH`.

**Reply by Atque** [reply]
> `ShowMySky-Qt6.dll` should be either near `stellarium.exe`, or in a location specified in `PATH`.

I suspected something like that. How do I generate it?

**Reply by gzotti** [reply]
Yes, config is a bit of a nuisance. Move the file D:\StelDev\GIT\build-stellarium-Desktop_Qt_6_4_1_MSVC2019_64bit-Release\_deps\showmysky-qt6-build\ShowMySky\ShowMySky-qt6.dll into D:\StelDev\GIT\stellarium

or add D:\StelDev\GIT\build-stellarium-Desktop_Qt_6_4_1_MSVC2019_64bit-Release\_deps\showmysky-qt6-build\ShowMySky\ to your PATH environment variable.

(adjust for your paths)

**Reply by Atque** [reply]
Thanks a lot, now it works. But wouldn’t it be easier if it automatically was placed in the same directory as the `stellarium.exe`?

**Reply by 10110111** [reply]
It would be easier to set the path to the library in `PATH` environment variable. Moving the libraries around via CMake is too fragile due to the difference between platforms.

**Reply by Atque** [reply]
Yes, that might be easier. Does the `ShowMySky-Qt6.dll` change with each release of CalcMySky?

**Reply by 10110111** [reply]
It may change in principle. There are no guarantees that it does or doesn't.
